From 6e3b8e407eda65a12c43afb9651fc284c6f154d6 Mon Sep 17 00:00:00 2001
From: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
Date: Wed, 26 Feb 2025 18:36:52 +0200
Subject: [PATCH] effect: Prevent QJSEngine garbage collecting EffectWindow

EffectWindow has no parent so QJSEngine can garbage collect an instance
if it's returned by a function marked with Q_INVOKABLE.

BUG: 466380
---
 src/effects.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/effects.cpp b/src/effects.cpp
index 3029b47e956..2e891cf3a8b 100644
--- a/src/effects.cpp
+++ b/src/effects.cpp
@@ -1993,6 +1993,8 @@ EffectWindowImpl::EffectWindowImpl(Window *window)
     : m_window(window)
     , m_windowItem(nullptr)
 {
+    QQmlEngine::setObjectOwnership(this, QQmlEngine::CppOwnership);
+
     // Deleted windows are not managed. So, when windowClosed signal is
     // emitted, effects can't distinguish managed windows from unmanaged
     // windows(e.g. combo box popups, popup menus, etc). Save value of the
-- 
GitLab

